include/mxnet/mem_mgr.h

```diff
@@ -72,18 +71,18 @@ class Block {
 class MemoryManager {
   Block* allocatedList_;
   Block* freeList_[FREELISTSIZE_];
+  Block* usedList_;
   std::mutex mutex_;
   public:
     static MemoryManager* Get();
     static std::shared_ptr<MemoryManager> _GetSharedRef();
-    ~MemoryManager();
     cudaError_t Malloc(void*& devptr, size_t size, int deviceIdx);
     cudaError_t Free(void* devptr, int deviceIdx);
     cudaError_t Memcpy(int deviceId, void* dst, 
                        const void* src, size_t count, enum cudaMemcpyKind kind);
     cudaError_t MemGetInfo(int deviceId, size_t* total, size_t* free);   
     bool TryAllocate(int deviceId, size_t size);
-    Block* findFirstFit(int idx, size_t size);
+    Block* findFirstFit(int idx, Block* prev, size_t size);
     Block* allocateBlock(size_t size);
     void splitAndPlace(Block* b, Block* prev, int idx, size_t size);
```

Comment:
Is there any specific reason why these block related APIs are public? If no, they should be private.